src/java/picard/sam/markduplicates/EstimateLibraryComplexity.java

```diff
@@ -127,13 +127,21 @@
             "group size would be approximately 10 reads.")
     public int MAX_GROUP_RATIO = 500;
 
+    @Option(doc = "Barcode SAM tag (ex. BC)", optional=true)
```

Comment:
We normally use the readgroup to get the library, but here (i think) the
library will be included in the RX tag after the dash (-). So if there's
only one library it will always be "-1" after the "gem" barcode.



On Mon, Sep 14, 2015 at 8:39 PM, Nils Homer <notifications@github.com>
wrote:

> In src/java/picard/sam/markduplicates/EstimateLibraryComplexity.java
> <https://github.com/broadinstitute/picard/pull/285#discussion_r39464157>:
>
> > @@ -127,13 +127,21 @@
> >              "group size would be approximately 10 reads.")
> >      public int MAX_GROUP_RATIO = 500;
> >
> > +    @Option(doc = "Barcode SAM tag (ex. BC)", optional=true)
>
> Looking at your NA12878 Bam file, the read group is empty, so using the BC
> tag in addition to BX/RX seems reasonable. What do you think @wheaton5
> <https://github.com/wheaton5>?
>
> Also, what should we do for reads that are missing an RX or BX tag? In
> the currently implementation, they would all be compared against each
> other, which is conservative.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/285/files#r39464157>.
>